auto_gptq/quantization/config.py

```diff
@@ -0,0 +1,222 @@
+import json
+import logging
+from dataclasses import dataclass, field, fields
+from os.path import isdir, join
+from typing import Optional
+
+from transformers.utils.hub import PushToHubMixin, cached_file
+
+
+logger = logging.getLogger(__name__)
+
+CHECKPOINT_FORMAT_FIELD = "checkpoint_format"
+CHECKPOINT_FORMAT_FIELD_COMPAT_MARLIN = "is_marlin_format"
+QUANT_METHOD_FIELD = "quant_method"
+QUANT_CONFIG_FILENAME = "quantize_config.json"
+
+# checkpoint formats
+class CHECKPOINT_FORMAT:
+    GPTQ = "gptq"
+    MARLIN = "marlin"
+    GEMM = "gemm"
+
+
+# quant methods
+class QUANT_METHOD:
+    GPTQ = "gptq"
+    AWQ = "awq"
+
+
+QUANT_METHOD_FORMAT_MAPPING = {
+    QUANT_METHOD.GPTQ: {
+        CHECKPOINT_FORMAT.GPTQ,
+        CHECKPOINT_FORMAT.MARLIN,
+    },
+    QUANT_METHOD.AWQ: {
+        CHECKPOINT_FORMAT.GEMM
+    }
```

Comment:
I am wondering why `CHECKPOINT_FORMAT.GPTQ` is not there? I recall the implem I did for AWQ was AWQ GEMM -> gptq format. Does that not work?

Reply:
Never tested that combo. Did even know it worked. Will test and change pending test result.

Reply:
@fxmarty  We tested the loading pure awq-gemm model quantized with autoawq on autogptq tip/main. This worked. 

But loading a  Yi6B model quantized by autogptq with groupsize=128,damp=0.005 failed when using setting `awq_gemm_checkpoint=True`

```python
INFO - Loading an AWQ model. This requires repacking the weights, and no repacking cached weight was found. Grab a coffee!
  0%|                                                            | 0/224 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/root/quantx/gptq_test.py", line 21, in <module>
    model_q = AutoGPTQForCausalLM.from_quantized(model_path, use_safetensors=True,
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/AutoGPTQ/auto_gptq/modeling/auto.py", line 140, in from_quantized
    return quant_func(
           ^^^^^^^^^^^
  File "/root/AutoGPTQ/auto_gptq/modeling/_base.py", line 1173, in from_quantized
    unpacked_qweight, unpacked_qzeros = unpack_awq(
                                        ^^^^^^^^^^^
  File "/root/AutoGPTQ/auto_gptq/modeling/_utils.py", line 603, in unpack_awq
    qdq_weight_T = weight * scale_mat - scale_zeros_mat.half()
                   ~~~~~~~^~~~~~~~~~~
RuntimeError: The size of tensor a (32768) must match the size of tensor b (4096) at non-singleton dimension 1
```

Reply:
@Qubitium Then let it be. I can have a look later.